### DISCUSSION TITLE
Refactoring vcpkg.cmake, would you welcome a PR?

### DISCUSSION THREAD

**Discussion by ptesavol** [microsoft/vcpkg#39068] [Ideas]
Hi

Currently [vcpkg.cmake](https://github.com/microsoft/vcpkg/blob/master/scripts/buildsystems/vcpkg.cmake) consists of over 900 lines of largely unstructured code that operates on file-scoped variables accessing them at will for both input and output. This makes it hard to understand what the code actually does, and also makes it hard to introduce changes to the code.

I would like to refactor the code into functions with clearly-defined responsibilities, inputs and outputs. (The functions could also be placed in separate files and included into the main vcpkg.cmake file, but this is not  the main goal). After the refactoring it would be much easier to implement new features such as [support for nested submodules/sub-projects] (https://github.com/microsoft/vcpkg/issues/15279). 

Before starting the refactoring work I would like to ask if you would you be ready to consider such a PR?

**Comment by AugP** [comment]
Thank you for your interest in improving vcpkg! I forwarded this to the rest of the maintainers to discuss at our next meeting. We should get back to you this week with an answer.

**Reply by AugP** [reply]
Update on this: we have discussed this, and we would only accept PRs that make refactoring changes if they pursue a clear user-impacting objective that we intend to reach. If your thought is to do this as a prerequisite to addressing issue #15279, we would first need to decide whether or not to resolve that issue in the discussion thread associated with it: https://github.com/microsoft/vcpkg/discussions/18122
